scripts/exgfs_wave_post_pnt.sh

```diff
@@ -624,7 +624,7 @@
 
   if [ ${CFP_MP:-"NO"} = "YES" ]; then nm=0; fi
 
-  if [ ${CFP_MP:-"NO"} = "YES" ]; then
+  if [ ${CFP_MP:-"NO"} = "YES" ] && [ "$DOBLL_WAV" = "YES" ]; then
```

Comment:
What was the issue here?  What if say, DOBLL_WAV=NO but DOSPC_WAVE=YES, do we get to line 630?

Reply:
If CFP_MP=YES we enter the if-block on [line 627](https://github.com/KateFriedman-NOAA/global-workflow/blob/issue178/scripts/exgfs_wave_post_pnt.sh#L627). Then, regardless of what DOBNDPNT_WAV equals, if DOBLL_WAV=NO and DOSPC_WAV=YES your nm counter coming out of that if-block will be "1". Then [wavenproc=1](https://github.com/KateFriedman-NOAA/global-workflow/blob/issue178/scripts/exgfs_wave_post_pnt.sh#L671) and since it's not greater than 1 it won't enter the [CFP_MP part of the next if-block on line 681](https://github.com/KateFriedman-NOAA/global-workflow/blob/issue178/scripts/exgfs_wave_post_pnt.sh#L681). So you set up your cmdfile with the leading nm value (what running with CFP_MP=YES needs for using --multi-prog or equivalent) but when the job goes to [the non-CFP_MP else](https://github.com/KateFriedman-NOAA/global-workflow/blob/issue178/scripts/exgfs_wave_post_pnt.sh#L691) it errors because of the leading nm # in the cmdfile. Essentially, we have to avoid coming out of a CFP_MP block with nm<2.

> What if say, DOBLL_WAV=NO but DOSPC_WAVE=YES, do we get to line 630?

If DOBLL_WAV=NO you won't go through the CFP_MP=YES part of the block, you'll go to the non-CFP_MP part on line 651 and get a cmdfile without the leading # on each line within for CFP_MP running.

Note, this is a highly specialized workaround for this bug. If additional DO*=YES blocks are added to that section that will avoid this situation and the check of DOBLL_WAV=YES can go away. A rework of this may be best in the future.